sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/caseWhenCoercion.sql

```diff
@@ -0,0 +1,200 @@
+--
+--   Licensed to the Apache Software Foundation (ASF) under one or more
+--   contributor license agreements.  See the NOTICE file distributed with
+--   this work for additional information regarding copyright ownership.
+--   The ASF licenses this file to You under the Apache License, Version 2.0
+--   (the "License"); you may not use this file except in compliance with
+--   the License.  You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+--   Unless required by applicable law or agreed to in writing, software
+--   distributed under the License is distributed on an "AS IS" BASIS,
+--   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+--   See the License for the specific language governing permissions and
+--   limitations under the License.
+--
+
+CREATE TEMPORARY VIEW t AS SELECT 1;
+
+SELECT CASE WHEN true THEN cast(1 as tinyint) ELSE cast(2 as tinyint) END FROM t;
```

Comment:
Just want to confirm whether these queries you added can be executed in Hive?

Reply:
@gatorsmile Two questions:
1. Hive doesn't have the `short` type, so can we remove the `short` type here?
2. Hive can't execute `CREATE TEMPORARY VIEW ...`, but can executor `CREATE TEMPORARY TABLE ...`, Do we add this feature to spark?

Reply:
In the future, we might do it for supporting the actual TEMP table. 

Yeah, please get rid of `short`.